### PR TITLE
Connector

### PR DESCRIPTION
**Pull request by Mittows**
- Installed connector-php submodule
- Refactored buildWuunderdata function in Data.php

### CODE REVIEW

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php [hunk 1]
```diff
@@ -335,31 +335,30 @@ public function buildWuunderData($infoArray, $order, $bookingToken)
...
-            'given_name' => Mage::getStoreConfig('wuunderconnector/connect/firstname'),
-            'locality' => Mage::getStoreConfig('wuunderconnector/connect/city'),
-            'phone_number' => Mage::getStoreConfig('wuunderconnector/connect/phone'),
-            'street_name' => Mage::getStoreConfig('wuunderconnector/connect/streetname'),
-            'house_number' => Mage::getStoreConfig('wuunderconnector/connect/housenumber'),
-            'zip_code' => Mage::getStoreConfig('wuunderconnector/connect/zipcode'),
-            'country' => Mage::getStoreConfig('wuunderconnector/connect/country')
-        );
+        $customerAdr = new \Wuunder\Api\Config\AddressConfig();
+
```
Comment (timoj) [inline app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Waar is business heen?

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php [hunk 2]
```diff
@@ -431,24 +430,32 @@ public function buildWuunderData($infoArray, $order, $bookingToken)
...
-            'weight' => $infoArray['weight'],
-            'delivery_address' => $customerAdr,
-            'pickup_address' => $webshopAdr,
-            'preferred_service_level' => $preferredServiceLevel,
-            'parcelshop_id' => $parcelshopId,
-            'source' => $sourceObj,
-            'redirect_url' => Mage::getUrl('adminhtml') . 'sales_order?label_order=' . $infoArray['order_id'],
-            'webhook_url' => Mage::getBaseUrl(Mage_Core_Model_Store::URL_TYPE_WEB) . 'wuunderconnector/webhook/call/order_id/' . $infoArray['order_id'] . "/token/" . $bookingToken
-        );
+        // return array(
```
Comment (timoj) [inline app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Remove comments

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php [hunk 3]
```diff
@@ -431,24 +430,32 @@ public function buildWuunderData($infoArray, $order, $bookingToken)
...
+        $bookingConfig->setPreferredServiceLevel($preferredServiceLevel);
+        $bookingConfig->setSource($sourceObj); 
+        $bookingConfig->setDeliveryAddress($customerAdr);
+        $bookingConfig->setPickupAddress($webshopAdr);
+        if ($parcelshopId) {
+            $bookingConfig->setParcelshopId($parcelshopId);
+        }
+
+        return $bookingConfig;
+
```
Comment (timoj) [inline app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Where did the redirect url and webhook go?

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php (hunk 1 above)
Comment (Mittows) [reply app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: I re-added it to the $customerAdr array.

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php (hunk 2 above)
Comment (Mittows) [reply app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Done

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php (hunk 3 above)
Comment (Mittows) [reply app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Added them to $bookingConfig after removing the comments.

File: app/code/community/Wuunder/WuunderConnector/Helper/Data.php [hunk 4]
```diff
@@ -202,55 +202,27 @@ public function processLabelInfo($infoArray)
...
-            // Create or update wuunder_shipment
-            if (!$this->saveWuunderShipment($infoArray)) {
-                $this->log("Something went wrong with saving wuunder shipment booking");
-                return array(
-                    'error' => true,
-                    'message' => 'Unable to create / update wuunder_shipment for order ' . $infoArray['order_id']
-                );
+        $connector = new Wuunder\Connector($apiKey, Mage::getStoreConfig('wuunderconnector/connect/testmode', $storeId) == 1);
+        $booking = $connector->createBooking();
+        $this->log($apiKey);
```
Comment (timoj) [inline app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Why log apiKey? old debug info?

Comment (Mittows) [reply app/code/community/Wuunder/WuunderConnector/Helper/Data.php]: Yes, removed the log line.

**Pull request merged**